src/app/common-components/header/header.component.ts

```diff
@@ -1,8 +1,28 @@
-import { Component } from "@angular/core";
+import { Component, OnInit } from "@angular/core";
+import { AuthService } from "src/app/core/services/auth.service";
+import { Router } from "@angular/router";
+import { User } from "src/app/login/_models/user";
 
 @Component({
-  selector: "app-header",
-  templateUrl: "./header.component.html",
-  styleUrls: ["./header.component.less"]
+    selector: "app-header",
+    templateUrl: "./header.component.html",
+    styleUrls: ["./header.component.less"]
 })
-export class HeaderComponent {}
+export class HeaderComponent implements OnInit {
+    isAuthenticated = false;
+    constructor(private authService: AuthService, private router: Router) {}
+
+    ngOnInit() {
+        this.authService.isAuthenticated$.subscribe(isAuthenticated => (this.isAuthenticated = isAuthenticated));
```

Comment:
you can just use async pipe in the template.
Or you have to unsubscribe your subscription.
Subscriptions are not cleaning by GC. They are left in memory. And will cause memory leaks.